source/extensions/filters/network/redis_proxy/router_impl.cc

```diff
@@ -121,14 +93,40 @@ RouteSharedPtr PrefixRoutes::upstreamPool(std::string& key) {
       key.erase(0, value->prefix().length());
     }
     if (!value->keyFormatter().empty()) {
-      formatKey(callbacks_, value, key, redis_key_formatter_command_);
+      std::string redis_key_formatter(value->keyFormatter());
+      formatKey(key, redis_key_formatter);
     }
     return value;
   }
 
   return catch_all_route_;
 }
 
+void PrefixRoutes::formatKey(std::string& key, std::string& redis_key_formatter) {
```

Comment:
Sorry, I originally suggested passing `redis_key_formatter` by reference because I thought we won't need modify the string content and can save the memory copy by pass-by-reference.

Since we are doing copy assignment at line 109, it shouldn't be much difference if we pass by value. So `std::string& redis_key_formatter` -> `std::string redis_key_formatter`.

Lines 96-97 can be changed to:
```
formatKey(key, value->keyFormatter());
```

Reply:
done. PTAL ?